web/app/models.py

```diff
@@ -226,7 +227,7 @@ def __str__(self):
 class TournamentPlayer(db.Model):
```

Comment:
I think you can make some simplifying assumptions.  Ideally, there's no reason why folks can't register as an AGA member immediately at the tournament site, thus creating a Player for them here, right?